### PR TITLE
some fixes for disable implicit conversions

### PR DESCRIPTION
**Pull request by dtaller**

### CODE REVIEW

File: src/care/KeyValueSorter.h [hunk 1]
```diff
@@ -1135,15 +1135,15 @@ class KeyValueSorter<T, RAJA::seq_exec> {
             std::sort(rawData, rawData + lsize, cmpKeys<T>);
 
             // Reallocate memory
-            if (m_keyValues) {
+            if (m_keyValues != nullptr) {
```
Comment (adayton1) [inline src/care/KeyValueSorter.h:1138]: The `if (ptr)` pattern should not need to be changed.

File: src/care/KeyValueSorter.h [hunk 2]
```diff
@@ -1206,7 +1206,7 @@ class KeyValueSorter<T, RAJA::seq_exec> {
       /// @return whether keys are allocated
       ///////////////////////////////////////////////////////////////////////////
       bool keysAllocated() const {
-         return m_keys ;
+         return m_keys != nullptr;
```
Comment (adayton1) [inline src/care/KeyValueSorter.h:1209]: This is the pattern that is fine to change. I'm fine with either what you have here or `return (bool) m_keys`. What you have here is probably actually better (easier to see what is going on).

Comment (dtaller) [reply src/care/KeyValueSorter.h:1209]: I kept it as != nullptr, sounds good

File: src/care/KeyValueSorter.h (hunk 1 above)
Comment (dtaller) [reply src/care/KeyValueSorter.h]: Reverted

**Pull request merged**